doc/debug-companion.md

```diff
@@ -1 +1,171 @@
 # ChakraCore Debugger Companion
+
+## Overview
+There's currently no turn-key way to debug ChakraCore-embedded scenarios.  It's possible to write a debugger using the JsDiag* API surface, but it's non-trivial to leverage any existing debugging tools.  This feature aims to provide an end-to-end flow for debugging a ChakraCore-embedded application VS Code.
+
+![Overview Diagram](./assets/companion-overview.png)  
+*Figure 1: High level overview of the components*
```

Comment:
We talked f2f about picture, but summary of comments:
  - it will be useful to understand where the debugger (e.g., VS Code) is in this picture.
  - Name "Debugging Client" is a little misleading here, as it can easily be interpreted as the "VS Code"/IDE thing.  
  - Name `Debug Companion` isn't really great.  Perhaps name of `DebugAdapter`
  - Name `Platform Implementation` isn't a little vague.  Maybe `Platform Support Library`.  Also, picture can be updated to include multiple overlapping boxes to indicate potentially multiple implementations (e.g., one on libuv, one on win32,...). 
  - DLL boundaries would be helpful, with specific names of the DLLs we plan to deliver.
  - what protocol is being used here to communicate between  debugger (e.g., VS Code) and the app.

Reply:
I've addressed a bunch of these comments, but I'm still playing with the names a bit.